### PR TITLE
Use `get_storage_at` procedures

### PR DESCRIPTION
**Pull request by i-norden**
Update to use https://github.com/cerc-io/ipld-eth-db/pull/127 for #6 and add some unit test cases to test the behavior when accounts or storage slots are removed from state.

### CODE REVIEW

File: statedb_test.go [hunk 1]
```diff
@@ -87,9 +93,70 @@ func TestSuite(t *testing.T) {
...
+		Nonce:       Account.Nonce,
+		CodeHash:    AccountCodeHash.String(),
+		StorageRoot: Account.Root.String(),
+		Removed:     false,
+	}))
+	require.NoError(t, insertStateCID(pool, stateModel{
+		BlockNumber: BlockNumber.Uint64() + 3,
+		BlockHash:   RandomHash3.String(),
+		LeafKey:     AccountLeafKey.String(),
+		CID:         RemovedNodeStorageCID,
```
Comment (roysc) [inline statedb_test.go]: nit 
```suggestion
		CID:         RemovedNodeStateCID,
```

File: statedb_test.go [hunk 2]
```diff
@@ -87,9 +93,70 @@ func TestSuite(t *testing.T) {
...
+		Diff:           true,
+		Value:          StoredValueRLP,
+		Removed:        false,
+	}))
+	require.NoError(t, insertStorageCID(pool, storageModel{
+		BlockNumber:    BlockNumber.Uint64() + 1,
+		BlockHash:      RandomHash.String(),
+		LeafKey:        AccountLeafKey.String(),
+		StorageLeafKey: StorageLeafKey.String(),
+		StorageCID:     RemovedNodeStateCID,
```
Comment (roysc) [inline statedb_test.go]:
```suggestion
		StorageCID:     RemovedNodeStorageCID,
```

File: statedb_test.go [hunk 3]
```diff
@@ -87,9 +93,70 @@ func TestSuite(t *testing.T) {
...
+		Diff:           true,
+		Value:          StoredValueRLP2,
+		Removed:        false,
+	}))
+	require.NoError(t, insertStorageCID(pool, storageModel{
+		BlockNumber:    BlockNumber.Uint64() + 3,
+		BlockHash:      RandomHash3.String(),
+		LeafKey:        AccountLeafKey.String(),
+		StorageLeafKey: StorageLeafKey.String(),
+		StorageCID:     RemovedNodeStateCID,
```
Comment (roysc) [inline statedb_test.go]:
```suggestion
		StorageCID:     RemovedNodeStorageCID,
```

File: statedb_test.go [hunk 4]
```diff
@@ -108,9 +175,33 @@ func TestSuite(t *testing.T) {
...
+		require.NoError(t, err)
+		require.Equal(t, &Account, acct2)
+
+		acct3, err := db.StateAccount(AccountLeafKey, RandomHash2)
+		require.NoError(t, err)
+		require.Equal(t, &Account, acct3)
+
+		acct4, err := db.StateAccount(AccountLeafKey, RandomHash3)
+		require.NoError(t, err)
+		require.Equal(t, (*types.StateAccount)(nil), acct4)
```
Comment (roysc) [inline statedb_test.go]:
```suggestion
		require.Nil(t, acct4)
```

Comment (roysc) [reply statedb_test.go]: Likewise for the other nils below

File: statedb_test.go [hunk 5]
```diff
@@ -180,8 +271,8 @@ func insertHeaderCID(db *pgxpool.Pool) error {
 	coinbase
 ) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14)`
 	_, err := db.Exec(testCtx, sql,
-		BlockNumber.Uint64(),
-		BlockHash.String(),
+		blockNumber,
+		blockHash,
 		BlockParentHash.String(),
 		HeaderCID.String(),
```
Comment (roysc) [inline statedb_test.go:281]: Let's also hardcode these to zero values so it's clear they're just filler values for now
```suggestion
		common.HexToHash("0").String(), "",
```

**Review by roysc — APPROVED** [review]
LGTM but have some nits and will note that we should add cases to check state & storage removal behavior, but there's already a TODO for that in the code, so 👍

File: statedb_test.go (hunk 5 above)
Comment (i-norden) [reply statedb_test.go:281]: Word! Thinking we should make variables for the incremented block numbers too

File: statedb_test.go (hunk 4 above)
Comment (i-norden) [reply statedb_test.go]: Thanks!

File: statedb_test.go (hunk 5 above)
Comment (i-norden) [reply statedb_test.go:281]: Gonna use `Header.UncleHash.String()`

**Comment by i-norden** [comment]
Merging this but have more fixes incoming

**Pull request merged**